product/src/test/java/org/itsallcode/openfasttrace/cli/ProcessOutputConsumer.java

```diff
@@ -0,0 +1,147 @@
+package org.itsallcode.openfasttrace.cli;
+
+import java.io.*;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+class ProcessOutputConsumer
+{
+    private static final Logger LOG = Logger.getLogger(ProcessOutputConsumer.class.getName());
+    private final Executor executor;
+    private final Process process;
+    private final ProcessStreamConsumer stdOutConsumer;
+    private final ProcessStreamConsumer stdErrConsumer;
+
+    ProcessOutputConsumer(final Process process)
+    {
+        this(createThreadExecutor(), process, new ProcessStreamConsumer("stdout"), new ProcessStreamConsumer("stderr"));
+    }
+
+    ProcessOutputConsumer(final Executor executor, final Process process,
+            final ProcessStreamConsumer stdOutConsumer, final ProcessStreamConsumer stdErrConsumer)
+    {
+        this.executor = executor;
+        this.process = process;
+        this.stdOutConsumer = stdOutConsumer;
+        this.stdErrConsumer = stdErrConsumer;
+    }
+
+    private static Executor createThreadExecutor()
+    {
+        return runnable -> {
+            final Thread thread = new Thread(runnable);
+            thread.setName(ProcessOutputConsumer.class.getSimpleName());
+            thread.setUncaughtExceptionHandler(new LoggingExceptionHandler());
+            thread.start();
+        };
+    }
+
+    void start()
+    {
+        executor.execute(() -> {
+            readStream(process.getInputStream(), stdOutConsumer);
+        });
+        executor.execute(() -> {
+            readStream(process.getErrorStream(), stdErrConsumer);
+        });
+    }
+
+    private void readStream(final InputStream stream, final ProcessStreamConsumer consumer)
+    {
+        try (final BufferedReader reader = new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8)))
+        {
+            String line = null;
```

Comment:
Strings are immutable. Does reusing it have any runtime benefit in this case? I would assume not, since the memory will be freshly allocated with each new assignment.